tests/std/tests/LWG4084_iostream_uppercase_inf_nan/test.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstddef>
+#include <limits>
+#include <sstream>
+#include <type_traits>
+
+using namespace std;
+
+template <class CharT, size_t NChar, size_t NWChar, enable_if_t<is_same_v<CharT, char>, int> = 0>
+constexpr const auto& choose_literal(const char (&s)[NChar], const wchar_t (&)[NWChar]) noexcept {
+    return s;
+}
+template <class CharT, size_t NChar, size_t NWChar, enable_if_t<is_same_v<CharT, wchar_t>, int> = 0>
+constexpr const auto& choose_literal(const char (&)[NChar], const wchar_t (&ws)[NWChar]) noexcept {
+    return ws;
+}
+
+#define STATICALLY_WIDEN(CharT, S) ::choose_literal<CharT>(S, L##S)
+
+template <class CharT, class F>
+void test() {
+    // LWG-4084 "std::fixed ignores std::uppercase"
+    {
+        auto s = (basic_ostringstream<CharT>{} << fixed << uppercase << numeric_limits<F>::infinity()).str();
+        assert(s == STATICALLY_WIDEN(CharT, "INF"));
+    }
+    {
+        auto s = (basic_ostringstream<CharT>{} << fixed << uppercase << numeric_limits<F>::quiet_NaN()).str();
+        assert(s == STATICALLY_WIDEN(CharT, "NAN"));
+    }
+    // also test other combinations

```

Comment:
No change requested: This is almost repetitive enough to be worth extracting into table-based test cases, and/or extracting `numeric_limits` into `constexpr` constants, but each of them is still a single line and it's easy enough to see what's being tested.